app/com/arpnetworking/kairos/config/MetricsQueryConfigImpl.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright 2020 Dropbox Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.arpnetworking.kairos.config;
+
+import com.arpnetworking.kairos.client.models.SamplingUnit;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableSet;
+import com.google.inject.Inject;
+import com.typesafe.config.Config;
+
+import java.util.List;
+import java.util.Locale;
+import java.util.Optional;
+import java.util.Set;
+import java.util.regex.Pattern;
+
+/**
+ * A {@link MetricsQueryConfig} that supports rollup blacklisting.
+ *
+ * @author Christian Briones (cbriones at dropbox dot com)
+ */
+public class MetricsQueryConfigImpl implements MetricsQueryConfig {
+
+    /**
+     * Construct a new {@code MetricsQueryImpl} by parsing a {@link Config}.
+     *
+     * @param configuration Play configuration to load from
+     */
+    @Inject
+    public MetricsQueryConfigImpl(final Config configuration) {
```

Comment:
I'd ordinarily want to separate out the configuration here but that would require a lot more boilerplate, especially since `buildBlacklistEntry` below has to crawl the config list. It also seems like configuration is directly injected elsewhere in the codebase.